Add known issue for audit logging with fitered fields (#84786)
elastic/elasticsearch: docs/reference/release-notes/8.1.asciidoc

Also see <<breaking-changes-8.1,Breaking changes in 8.1>>.

[[known-issues-8.1.0]]
[float]
=== Known issues

* If audit logging is configured to `emit_request_body`, it can prevent some APIs from working.
+
--
In 8.1.0, a bug {es-issue}84784[#84784] prevents APIs using requests with filtered fields from working
when request body auditing is enabled. Using these APIs can result into a `401` HTTP
error with a `null_pointer_exception` being the underlying cause.

> 

Author: Thanks I consolidated the sentences with some minor edits.

Audit logging can be configured to print the request body of incoming requests.
Certain requests can contain sensitive information, e.g. a password, that must not
be logged. Fields containing such information are filtered out before logging.
The REST APIs that use requests with filtered fields are:

* <<security-api-change-password>>
* <<cluster-nodes-reload-secure-settings>>
* <<security-api-put-user>>
* <<security-api-get-token>>
* <<security-api-invalidate-token>>
* <<security-api-grant-api-key>>
* <<watcher-api-put-watch>>
* <<watcher-api-execute-watch>>
* <<security-api-saml-authenticate>>
* <<security-api-oidc-authenticate>>
* <<docs-reindex,Reindex API>>

Please note `emit_request_body` defaults to `false`. Also note request body is only logged for
following events (not all of these events are enabled by default):

* `authentication_success`
* `authentication_failed`
* `realm_authentication_failed`
* `tampered_request`
* `run_as_denied`
* `anonymous_access_denied`
--

> 

Reviewer: We don't give a clear workaround here. Should state that you can avoid the errors by setting `xpack.security.audit.logfile.events.emit_request_body` to `false` and include a related snippet?

```
[source,console]
----
PUT /_cluster/settings
{
  "persistent": {
    "xpack.security.audit.logfile.events.emit_request_body": false
  }
}
----

Author: Thanks. I add a paragrah for the workaround.

Also see <<breaking-changes-8.1,Breaking changes in 8.1>>.

[[known-issues-8.1.0]]
[float]
=== Known issues

* If audit logging is configured to `emit_request_body`, it can result into `NullPointerException` for
requests that have filtered fields.
+
--
Audit logging can be configured to print the request body of incoming requests.
Certain requests can contain sensitive information, e.g. password, that must not
be logged. Fields containing such information are filtered out before logging.
The REST APIs that use requests with filtered fields are:

* <<security-api-change-password>>
* <<cluster-nodes-reload-secure-settings>>
* <<security-api-put-user>>
* <<security-api-get-token>>
* <<security-api-invalidate-token>>
* <<security-api-grant-api-key>>
* <<watcher-api-put-watch>>
* <<watcher-api-execute-watch>>
* <<security-api-saml-authenticate>>
* <<security-api-oidc-authenticate>>
* <<docs-reindex,Reindex API>>

> In 8.1, a bug {es-issue}84784[#84784] prevents request with filtered fields work

Reviewer 2: ```suggestion
In 8.1, a bug {es-issue}84784[#84784] prevents request with filtered fields from working
```

correctly with auditing request body. Using above APIs can result into a `401` HTTP

> error with a `null_pointer_exception` being the underlying cause.

Reviewer 2: This paragraph should be before the list of APIs. We need to explain the impact of the bug before listing the affected APIs.

Author: Updated.

Please note `emit_request_body` defaults to `false`. Also note request body is only logged for

> following events (most of them are not enabled by default either):

Reviewer 2: ```suggestion
following events (not all of these events are enabled by default):
```
